client/utils/vocabularies.ts

```diff
@@ -28,15 +28,14 @@ export function getVocabularyItemFieldTranslated(
 export function getVocabularyItemNames<T>(
     selected: Array<T>,
     options: Array<T>,
-    valueField: keyof T,
+    valueField: string,
     nameField: keyof T,
     language: string
 ): Array<string> {
     if (!selected?.length) {
         return [];
     }
-
-    const values = selected.map((item) => item[valueField]);
+    const values = valueField === '_id' ? selected : selected.map((item) => item[valueField]);
```

Comment:
It is possible that we'd like to map items from `selected` using `valueField=='_id'`, which would be broken with these changes.

Maybe instead, if the types were changed to `valueField: keyof T | undefined` and check here if `valueField == null` to determine if we're to map `selected` or keep them as they are.

Reply:
> It is possible that we'd like to map items from `selected` using `valueField=='_id'`, which would be broken with these changes.
> 
I am not sure, but IMO its not possible because in `selected` we got only item `_id` not whole item

> Maybe instead, if the types were changed to `valueField: keyof T | undefined` and check here if `valueField == null` to determine if we're to map `selected` or keep them as they are.

I am little bit confusing in that. Would you kindly elaborate?

Reply:
This utility is designed to provide the translated values based on a given array of items.
By default, we're using `qcode` to get the list of values but it is plausible that we could use `_id` if given an array of objects (separate to the usage by Agendas).

Therefor, we should support supplying `valueField='_id'` while `selected` is an array of Objects.
We can support the above use case as well as an array of strings if this function is supplied `valueField=null`